bgpd/bgp_fsm.c

```diff
@@ -1206,6 +1205,259 @@ static void bgp_update_delay_process_status_change(struct peer *peer)
 	}
 }
 
+static bool bgp_gr_check_all_eors(struct bgp *bgp, afi_t afi, safi_t safi)
+{
+	struct listnode *node, *nnode;
+	struct peer *peer = NULL;
+
+	if (BGP_DEBUG(graceful_restart, GRACEFUL_RESTART))
+		zlog_debug("%s: Checking all peers for EOR receipt for %s",
+			   bgp->name_pretty, get_afi_safi_str(afi, safi, false));
+
+	for (ALL_LIST_ELEMENTS(bgp->peer, node, nnode, peer)) {
+		if (BGP_DEBUG(graceful_restart, GRACEFUL_RESTART))
+			zlog_debug("....examining peer %s status %s flags 0x%" PRIx64
+				   " af_sflags 0x%x",
+				   peer->host,
+				   lookup_msg(bgp_status_msg,
+					      peer->connection->status, NULL),
+				   peer->flags, peer->af_sflags[afi][safi]);
+		if (!CHECK_FLAG(peer->flags, PEER_FLAG_CONFIG_NODE) ||
+		    CHECK_FLAG(peer->flags, PEER_FLAG_SHUTDOWN) ||
+		    !CHECK_FLAG(peer->flags, PEER_FLAG_GRACEFUL_RESTART))
+			continue;
+
+		if (!CHECK_FLAG(peer->af_sflags[afi][safi],
+				PEER_STATUS_GR_WAIT_EOR))
+			continue;
+
+		if (!CHECK_FLAG(peer->af_sflags[afi][safi],
+				PEER_STATUS_EOR_RECEIVED)) {
+			if (BGP_DEBUG(graceful_restart, GRACEFUL_RESTART))
+				zlog_debug(
+					".... EOR still awaited from this peer for this AFI/SAFI");
```

Comment:
`for this AFI/SAFI` does not say anything, TBH for an operator. Can we print an actual values?